android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -1770,14 +1783,15 @@ public void setFullscreen(boolean fullscreen) {
                         | SYSTEM_UI_FLAG_FULLSCREEN;
             }
             eventEmitter.fullscreenWillPresent();
-            decorView.setSystemUiVisibility(uiOptions);
```

Comment:
You cannot do that here, it is really changing the behvior ... 
if user want to use fullscreen props (it will trigger this code) but have its own player UI, it will break the behavior...
The issue I describe should be reproducible in the basic sample !
(note for later it would be interesting to have a button in sample to trigger controls)

Reply:
I do not understand. decorView.setSystemUiVisibility(uiOptions) is called inside `post()`. Otherwise, system status bar does not have fade in and out animation

Reply:
See https://github.com/react-native-video/react-native-video/blob/a79c30eaba277db4cd13b3515567c17b7a09ee58/android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java#L1794

Reply:
I would to test on sample (see my last comment :) )
I will try to explain again:
- I have my own player overlay for playback control
- now fullscreen prop will launch the dialog.
- I assume (just by reading code) that when the dialog will be open, my own player control will be hidden

This use case shall be testable in basic sample (there is a fullsceen butotn available).
I can test once the build issue is fixed.

Thank you

Reply:
notice that I think you just need to duplicate/refactor this code, the phylosophie is ok for me

Reply:
As for the duplicating the code, you mean this  
```java
if (isFullscreen) {
    if (Util.SDK_INT >= 19) { // 4.4+
        uiOptions = SYSTEM_UI_FLAG_HIDE_NAVIGATION
                | SYSTEM_UI_FLAG_IMMERSIVE_STICKY
                | SYSTEM_UI_FLAG_FULLSCREEN;
    } else {
        uiOptions = SYSTEM_UI_FLAG_HIDE_NAVIGATION
                | SYSTEM_UI_FLAG_FULLSCREEN;
    }
    eventEmitter.fullscreenWillPresent();
    fullScreenPlayerView.show();
    post(() -> {
      decorView.setSystemUiVisibility(uiOptions)
      eventEmitter.fullscreenDidPresent();
    };
} else {
    uiOptions = View.SYSTEM_UI_FLAG_VISIBLE;
    eventEmitter.fullscreenWillDismiss();
    fullScreenPlayerView.dismiss();
    post(() -> {
      decorView.setSystemUiVisibility(uiOptions);
      eventEmitter.fullscreenDidDismiss();
    });
}
```

Reply:
> I have my own player overlay for playback control
now fullscreen prop will launch the dialog.
I assume (just by reading code) that when the dialog will be open, my own player control will be hidden

Yes you are correct. I have found I have not handled a scenario `<Video fullscreen=true />` and it cause the runtime error. 

In the basic example when I use `<Video fullscreen=true />` and then I use hardware back to exit. the fullscreen props does not sync with the native state. If you use the overlay fullscreen, it will have 3 behaviour (js fullscreen and native fullscreen and non-fullscreen)

I do not know what it should behave. I want to see how iOS does but I cannot build

Reply:
![Kapture 2022-09-08 at 00 45 50](https://user-images.githubusercontent.com/5212215/189065642-51ad8a05-872d-442a-855c-ae37d6c6f3d1.gif)

Reply:
Great, you got my point 👍
You should do a little bit more of refactoring.

Create a function : toggleDecoration which will handle decoration management
Call this function from setFullscreen and from the button handler

It should be ok like this ?!

Reply:
Without affect this existing behaviour

Why dont do this 

```js
if (isFullscreen) {
    if (Util.SDK_INT >= 19) { // 4.4+
        uiOptions = SYSTEM_UI_FLAG_HIDE_NAVIGATION
                | SYSTEM_UI_FLAG_IMMERSIVE_STICKY
                | SYSTEM_UI_FLAG_FULLSCREEN;
    } else {
        uiOptions = SYSTEM_UI_FLAG_HIDE_NAVIGATION
                | SYSTEM_UI_FLAG_FULLSCREEN;
    }
    eventEmitter.fullscreenWillPresent();
    if (controls) {
        fullScreenPlayerView.show();
    }
    eventEmitter.fullscreenDidPresent();
  } else {
    uiOptions = View.SYSTEM_UI_FLAG_VISIBLE;
    eventEmitter.fullscreenWillDismiss();
    if (controls) {
        fullScreenPlayerView.dismiss();
    }
    eventEmitter.fullscreenDidDismiss();
  }
  post(() -> decorView.setSystemUiVisibility(uiOptions));
}
```

Reply:
@wood1986 OK that also works for me !

Reply:
I fixed